src/lib/snack-bar/base-snack-bar.ts

```diff
@@ -0,0 +1,12 @@
+import {MdSnackBarRef} from './snack-bar-ref';
+
+
+export class BaseSnackBarContent<T> {
```

Comment:
Custom components used for snack bar content would also need to inherit from BaseSnackBarContent as it defines the MdSnackBarRef attributes on the component as well as a dismiss method to dismiss the snack bar.